tests/netebpfext_unit/netebpfext_unit.cpp

```diff
@@ -913,9 +937,21 @@ _Must_inspect_result_ ebpf_result_t
 netebpfext_unit_invoke_sock_ops_program(
     _In_ const void* client_binding_context, _In_ const void* context, _Out_ uint32_t* result)
 {
-    auto client_context = (test_sock_ops_client_context_t*)client_binding_context;
     UNREFERENCED_PARAMETER(context);
-    *result = client_context->sock_ops_action;
+    auto client_context = (test_sock_ops_client_context_t*)client_binding_context;
+    int action = client_context->sock_ops_action;
+
+    if (action == SOCK_OPS_TEST_ACTION_ROUND_ROBIN) {
+        // If the action is round robin, decide the action based on the local port number.
+        bpf_sock_ops_t* sock_ops_context = (bpf_sock_ops_t*)context;
+        action = _get_sock_ops_action(sock_ops_context->local_port);
+    }
+
+    if (action == SOCK_OPS_TEST_ACTION_PERMIT) {
+        *result = EBPF_SUCCESS;
```

Comment:
I think this is not correct. `result` should contain one of the error codes that sock_ops program can return.

Also, I think we should add a variation that the program invocation itself failed, which means this function returned a failure return code.

Reply:
Yes it seems BPF program return value is handled as below:

`(result == 0) ? FWP_ACTION_PERMIT : FWP_ACTION_BLOCK;`

So the program can return 0 or -1.